src/sentry/api/endpoints/group_events.py

```diff
@@ -36,47 +36,71 @@ def get(self, request, group):
         :auth: required
         """
 
-        events = Event.objects.filter(
-            group_id=group.id,
-        )
-
-        query = request.GET.get('query')
-
-        if query:
+        def respond(queryset):
+            return self.paginate(
+                request=request,
+                queryset=queryset,
+                order_by='-datetime',
+                on_results=lambda x: serialize(x, request.user),
+                paginator_cls=DateTimePaginator,
+            )
+
+        events = Event.objects.filter(group_id=group.id)
+
+        try:
+            environment = self._get_environment_from_request(
+                request,
+                group.project.organization_id,
+            )
+        except Environment.DoesNotExist:
+            return respond(events.none())
+
+        raw_query = request.GET.get('query')
+
+        if raw_query:
             try:
-                query_kwargs = parse_query(group.project, query, request.user)
+                query_kwargs = parse_query(group.project, raw_query, request.user)
             except InvalidQuery as exc:
                 return Response({'detail': six.text_type(exc)}, status=400)
+            else:
+                query = query_kwargs.pop('query', None)
+                tags = query_kwargs.pop('tags', {})
+                assert not query_kwargs, 'unexpected query parameters arguments: {!r}'.format(
```

Comment:
probably shouldn't do assertion this as it changes behavior afaict (it will error where it previously wouldnt have)

Reply:
Removed here: a28a0360e45cd09b22eceb89cb243af2d84f1bcc